src/main/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetTransferServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package edu.cornell.kfs.module.cam.document.service.impl;
+
+import java.util.regex.PatternSyntaxException;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.module.cam.businessobject.Asset;
+import org.kuali.kfs.module.cam.businessobject.AssetGlpeSourceDetail;
+import org.kuali.kfs.module.cam.businessobject.AssetPayment;
+import org.kuali.kfs.module.cam.document.AssetTransferDocument;
+import org.kuali.kfs.module.cam.document.service.impl.AssetTransferServiceImpl;
+import org.kuali.rice.coreservice.api.parameter.EvaluationOperator;
+import org.kuali.rice.coreservice.api.parameter.Parameter;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.cornell.kfs.module.cam.CuCamsConstants;
+
+public class CuAssetTransferServiceImpl extends AssetTransferServiceImpl {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuAssetTransferServiceImpl.class);
+
+    protected ParameterService parameterService;
+
+    /**
+     * Overridden to forcibly set the sub-account number on the generated postable object to null, if the account number
+     * is flagged as being one that should not have a sub-account number defined on related source details.
+     * The "ASSET_TRANSFER_PLANT_ACCOUNTS_FOR_SUB_ACCOUNTS" parameter governs whether such clearing behavior should be applied;
+     * see shouldPreserveSubAccount() for details.
+     * 
+     * @see org.kuali.kfs.module.cam.document.service.impl.AssetTransferServiceImpl#createAssetGlpePostable(
+     * org.kuali.kfs.module.cam.document.AssetTransferDocument, org.kuali.kfs.coa.businessobject.Account,
+     * org.kuali.kfs.module.cam.businessobject.AssetPayment, boolean, org.kuali.kfs.module.cam.document.service.impl.AssetTransferServiceImpl.AmountCategory)
+     */
+    @Override
+    protected AssetGlpeSourceDetail createAssetGlpePostable(
+            AssetTransferDocument document, Account plantAccount, AssetPayment assetPayment, boolean isSource, AmountCategory amountCategory) {
+        AssetGlpeSourceDetail postable = super.createAssetGlpePostable(document, plantAccount, assetPayment, isSource, amountCategory);
+        clearSubAccountIfNecessary(postable);
+        return postable;
+    }
+
+    /**
+     * Clears out the sub-account number on the source detail if the account number matches a particular pattern.
+     * In the event that a badly-formatted pattern causes an exception while matching, the source detail will be left as-is.
+     * 
+     * This was separated into its own method for unit testing convenience. 
+     * 
+     * @param postable The GLPE source detail whose sub-account number may need to be cleared based on account number.
+     */
+    protected void clearSubAccountIfNecessary(AssetGlpeSourceDetail postable) {
+        try {
+            if (!shouldPreserveSubAccount(postable)) {

```

Comment:
I'm generally not a fan of negative logic since it seems to require more thought to process. Would it be possible to invert this to something like shouldClearSubAccount()?